403 forbidden: package.json@^2.0.1

**Describe the bug**
I think there may be a discrepant behavior between verdaccio and official npm repo when it comes to `package.json` package. 

**To Reproduce**
Steps to reproduce the behavior:
1. `npm set registry https://registry.npmjs.org`
2. `npm install package.json`
3. `ls node_modules | grep package.json`
   ```git-package-json
   package-json
   package.json
   package-json-patch```
   ...
4. rm -r *
5. `npm set registry https:{your verdaccio repo}`
6. `npm install package.json`
7. error message is:
   ```npm ERR! code E403
   npm ERR! 403 Forbidden: package.json@latest

   npm ERR! A complete log of this run can be found in:
   npm ERR!     /home/{whoami}/.npm/_logs/2018-11-19T06_46_27_794Z-debug.log```


**Expected behavior**
Should install package.json package

**Configuration File (cat ~/.config/verdaccio/config.yaml)** 
```
#
# This is the default config file. It allows all users to do anything,
# so don't use it on production systems.
#
# Look here for more config file examples:
# https://github.com/verdaccio/verdaccio/tree/master/conf
#

url_prefix: https://npm.myrepo.com/
# path to a directory with all packages
storage: ./storage
# path to a directory with plugins to include
plugins: ./plugins

web:
  # WebUI is enabled as default, if you want disable it, just uncomment this line
  #enable: false
  title: My private repo

auth:
  htpasswd:
    file: ./htpasswd
    # Maximum amount of users allowed to register, defaults to "+inf".
    # You can set this to -1 to disable registration.
    max_users: 1 

# a list of other known repositories we can talk to
uplinks:
  npmjs:
    url: https://registry.npmjs.org/

packages:
  '@*/*':
    # scoped packages
    access: $all
    publish: $authenticated
    proxy: npmjs

  'private-*':
    access: $all
    publish: mybuild
    proxy: npmjs

  '**':
    # allow all users (including non-authenticated users) to read and
    # publish all packages
    #
    # you can specify usernames/groupnames (depending on your auth plugin)
    # and three keywords: "$all", "$anonymous", "$authenticated"
    access: $all

    # allow all known users to publish packages
    # (anyone can register by default, remember?)
    publish: mybuild 

    # if package is not available locally, proxy requests to 'npmjs' registry
    proxy: npmjs

# To use `npm audit` uncomment the following section
middlewares:
  audit:
    enabled: true

# log settings
logs:
  - { type: rotating-file, format: json, path: /home/verdaccio/verdaccio/log/log.json, level: info, options: { period: 2d } }
  #- {type: stdout, format: pretty, level: http}
  #- {type: file, path: verdaccio.log, level: info}

# listen address (or a port)
listen: 0.0.0.0:4873
```

**Debugging output**
This is log info grep'ing package.json on the repo:
```
{"name":"verdaccio","hostname":"xxx","pid":23352,"sub":"in","level":30,"req":{"method":"GET","url":"/package.json","headers":{"host":"npm.myrepo.com:443","x-real-ip":"x.x.x.x","x-forwarded-for":"x.x.x.x","x-forwarded-proto":"https","connection":"close","user-agent":"npm/6.2.0 node/v10.9.0 linux x64","npm-in-ci":"false","npm-scope":"","npm-session":"d510c4c313a46e56","referer":"install","pacote-req-type":"packument","pacote-pkg-id":"registry:manifest","accept":"application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*","authorization":"<Classified>","accept-encoding":"gzip,deflate"},"remoteAddress":"127.0.0.1","remotePort":33900},"ip":"127.0.0.1","msg":"127.0.0.1 requested 'GET /package.json'","time":"2018-11-19T06:39:01.767Z","v":0}
{"name":"verdaccio","hostname":"xxx","pid":23352,"sub":"in","level":35,"request":{"method":"GET","url":"/package.json"},"user":"mybuild","remoteIP":"x.x.x.x via 127.0.0.1","status":403,"error":"invalid package","bytes":{"in":0,"out":33},"msg":"403, user: mybuild(47.152.46.150 via 127.0.0.1), req: 'GET /package.json', error: invalid package","time":"2018-11-19T06:39:01.771Z","v":0}
{"name":"verdaccio","hostname":"xxx","pid":23352,"sub":"in","level":30,"req":{"method":"GET","url":"/package.json","headers":{"host":"npm.myrepo.com:443","x-real-ip":"x.x.x.x","x-forwarded-for":"x.x.x.x","x-forwarded-proto":"https","connection":"close","user-agent":"npm/6.2.0 node/v10.9.0 linux x64","npm-in-ci":"false","npm-scope":"","npm-session":"4b3f43c621f4e30d","referer":"install package.json","pacote-req-type":"packument","pacote-pkg-id":"registry:manifest","accept":"application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*","authorization":"<Classified>","accept-encoding":"gzip,deflate"},"remoteAddress":"127.0.0.1","remotePort":33916},"ip":"127.0.0.1","msg":"127.0.0.1 requested 'GET /package.json'","time":"2018-11-19T06:46:27.957Z","v":0}
{"name":"verdaccio","hostname":"xxx","pid":23352,"sub":"in","level":35,"request":{"method":"GET","url":"/package.json"},"user":"mybuild","remoteIP":"x.x.x.x via 127.0.0.1","status":403,"error":"invalid package","bytes":{"in":0,"out":33},"msg":"403, user: mybuild(x.x.x.x via 127.0.0.1), req: 'GET /package.json', error: invalid package","time":"2018-11-19T06:46:27.978Z","v":0}
```

**Additional context**
n/a

--- a/src/lib/local-storage.js
+++ b/src/lib/local-storage.js
@@ -409,7 +409,7 @@ class LocalStorage implements IStorage {
       _transform.apply(uploadStream, arguments);
     };
 
-    if (name === pkgFileName || name === '__proto__') {
+    if (name === '__proto__') {
       process.nextTick(() => {
         uploadStream.emit('error', ErrorCode.getForbidden());
       });

--- a/src/lib/utils.js
+++ b/src/lib/utils.js
@@ -75,7 +75,6 @@ function validateName(name: string): boolean {
     name.charAt(0) === '-' || // "-" is reserved by couchdb
     name === 'node_modules' ||
     name === '__proto__' ||
-    name === 'package.json' ||
     name === 'favicon.ico'
   );
 }

--- a/test/functional/sanity/security.js
+++ b/test/functional/sanity/security.js
@@ -9,7 +9,7 @@ export default function(server) {
     });
 
     test('should fails on fetch bad pkg #1', () => {
-      return server.getPackage('package.json')
+      return server.getPackage('__proto__')
         .status(HTTP_STATUS.FORBIDDEN)
         .body_error(/invalid package/);
     });
@@ -31,8 +31,8 @@ export default function(server) {
         });
     });
 
-    test('should fails and do not return package.json as an attachment', () => {
-      return server.request({uri: '/testpkg-sec/-/package.json'})
+    test('should fails and do not return __proto__ as an attachment', () => {
+      return server.request({uri: '/testpkg-sec/-/__proto__'})
         .status(HTTP_STATUS.FORBIDDEN)
         .body_error(/invalid filename/);
     });
@@ -49,7 +49,7 @@ export default function(server) {
     });
 
     test('should fails on fetch silly things - writing #1', () => {
-      return server.putTarball('testpkg-sec', 'package.json', '{}')
+      return server.putTarball('testpkg-sec', '__proto__', '{}')
         .status(HTTP_STATUS.FORBIDDEN)
         .body_error(/invalid filename/);
     });

--- a/test/unit/api/local-storage.spec.js
+++ b/test/unit/api/local-storage.spec.js
@@ -344,7 +344,7 @@ describe('LocalStorage', () => {
           stream.on('error', (err) => {
             expect(err).not.toBeNull();
             expect(err.statusCode).toEqual(HTTP_STATUS.CONFLICT);
-            expect(err.message).toMatch(/this package is already present/); 
+            expect(err.message).toMatch(/this package is already present/);
           });
           stream.on('success', function(){
             expect(spy).toHaveBeenCalled();
@@ -385,7 +385,7 @@ describe('LocalStorage', () => {
         });
 
         test('should fails on abort on add a new tarball', (done) => {
-          const stream = storage.addTarball('package.json', `${pkgName}-fails-add-tarball-1.0.4.tgz`);
+          const stream = storage.addTarball('__proto__', `${pkgName}-fails-add-tarball-1.0.4.tgz`);
           stream.abort();
           stream.on('error', function(err) {
             expect(err).not.toBeNull();

--- a/test/unit/api/utils.spec.js
+++ b/test/unit/api/utils.spec.js
@@ -59,10 +59,6 @@ describe('Utilities', () => {
       assert(validate('JSONStream'));
     });
 
-    test('no package.json', () => {
-      assert(!validate('package.json'));
-    });
-
     test('no path seps', () => {
       assert(!validate('some/thing'));
       assert(!validate('some\\thing'));

